packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { SelectControl } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+function __experimentalBlockVariationTransforms( {
+	blockName,
+	selectedBlockClientId,
+} ) {
+	const [ variationsSelectValue, setVariationsSelectValue ] = useState( '' );
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+	const { variations } = useSelect(
+		( select ) => {
+			const { getBlockVariations } = select( 'core/blocks' );
+			const scope = 'transform';
+			const _variations = getBlockVariations(
+				blockName,
+				scope
+			).filter( ( variation ) => variation.scope?.includes( scope ) );
+			return {
+				variations: _variations,
+			};
+		},
+		[ blockName ]
+	);
+	if ( ! variations.length ) return null;
+
+	const selectOptions = [
+		{ value: '', label: __( 'Select a variation' ) },
+		...variations.map( ( variation ) => ( {
+			value: variation.name,
+			label: variation.title || 'no title??? change me',
```

Comment:
Is it possible that a variation wouldn't have a title? I thought it is mandatory.

Reply:
I have to check that again but I saw in `Search` variation that it doesn't have one.. https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/search/variations.js

Reply:
I'm sure that the variation for search inherits the title from the block type then.